src/QsCompiler/Compiler/Templates/CppInterop.cs

```diff
@@ -0,0 +1,193 @@
+                    DataType.IntegerType => "vector<int64_t>",
+                    DataType.DoubleType => "vector<double_t>",
+                    DataType.PauliType => "std::vector<PauliId>",
+                    DataType.RangeType => "vector<RangeTuple>",
+                    DataType.ResultType => "vector<char>",
+                    DataType.StringType => "vector<string>",
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string? CppVarInitialValue()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "InteropFalseAsChar",
+                DataType.IntegerType => "0",
+                DataType.DoubleType => "0.0",
+                DataType.PauliType => "PauliId::PauliId_I",
+                DataType.RangeType => null,
+                DataType.ResultType => "InteropResultZeroAsChar",
+                DataType.StringType => null,
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => null,
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => null,
+                    DataType.RangeType => null,
+                    DataType.ResultType => null,
+                    DataType.StringType => null,
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+
+        public string? TransformationType()
+        {
+            return this.Type switch
+            {
+                DataType.BoolType => "BoolAsCharMap",
+                DataType.IntegerType => null,
+                DataType.DoubleType => null,
+                DataType.PauliType => "PauliMap",
+                DataType.RangeType => null,
+                DataType.ResultType => "ResultAsCharMap",
+                DataType.StringType => null,
+                DataType.ArrayType => this.ArrayType switch
+                {
+                    DataType.BoolType => "BoolAsCharMap",
+                    DataType.IntegerType => null,
+                    DataType.DoubleType => null,
+                    DataType.PauliType => "PauliMap",
+                    DataType.RangeType => null,
+                    DataType.ResultType => "ResultAsCharMap",
+                    DataType.StringType => null,
+                    _ => throw new NotSupportedException($"Unsupported array type {this.Type}")
+                },
+                _ => throw new NotSupportedException($"Unsupported argument type {this.Type}")
+            };
+        }
+    }
+
+    public class EntryPointOperationCpp : EntryPointOperation
```

Comment:
This class should also be `internal` since we don't want to expose its functionality outside its binary.

Reply:
done

Reply:
done